saleor/discount/tasks.py

```diff
@@ -227,12 +227,12 @@ def decrease_voucher_code_usage_of_draft_orders(channel_id: int):
         .filter(code__in=codes)
         .values_list("pk", flat=True)
     )
-    decrease_voucher_codes_usage_task.delay(list(voucher_code_ids))
+    decrease_voucher_codes_usage_task.delay(list(voucher_code_ids), list(codes))
 
 
 @app.task
 @allow_writer()
-def decrease_voucher_codes_usage_task(voucher_code_ids):
+def decrease_voucher_codes_usage_task(voucher_code_ids, codes):
     # Batch of size 1000 takes ~1sec and consumes ~20mb at peak
```

Comment:
Did you check if the comment here is still valid?

Side note: I also wonder if using `only` in this task still makes sense (not quite sure looking just at the code), so I will also point, that it may be worth double-checking the generated SQL's in this task if there will be high memory increase after changes.

Reply:
The sql queries are not changed. This PR pass the result of the query which was already evaluated and there is no significant impact on the task performance